doc/source/guide/in_depth/numpy_guide.rst

```diff
@@ -0,0 +1,124 @@
+
+To cast a NumPy array to an ODL vector, you simply need to call the `LinearSpace.element` method in an appropriate space::
+
+   >>> r3 = odl.Rn(3)
+   >>> arr = np.array([1, 2, 3])
+   >>> x = r3.element(arr)
+
+If the data type and storage methods allow it, the vector simply wraps the underlying array using a `view
+<http://docs.scipy.org/doc/numpy/glossary.html#term-view>`_::
+   
+   >>> float_arr = np.array([1.0, 2.0, 3.0])
+   >>> x = r3.element(float_arr)
+   >>> x.data is float_arr
+   True
+
+Casting ODL vectors to NumPy arrays can be done in two ways, either through the member function `NtuplesBaseVector.asarray`, or using `numpy.asarray`. These are both optimized and if possible return a view::
+
+   >>> x.asarray()
+   array([1.0, 2.0, 3.0])
+   >>> np.asarray(x)
+   array([1.0, 2.0, 3.0])
+
+These methods work with any ODL vector represented by an array. For example, in discretizations, a two-dimensional array can be used::
+
+   >>> X = odl.uniform_discr([0, 0], [1, 1], [3, 3])
+   >>> arr = np.array([[1, 2, 3],
+   ...                 [4, 5, 6],
+   ...                 [7, 8, 9]])
+   >>> x = X.element(arr)
+   >>> x.asarray()
+   array([[1.0, 2.0, 3.0],
+          [4.0, 5.0, 6.0],
+          [7.0, 8.0, 9.0]])
+
+Using ODL vectors with NumPy functions
+--------------------------------------
+It is often useful to use NumPy functions with ODL vectors. For universal functions (`ufuncs
+<http://docs.scipy.org/doc/numpy/reference/ufuncs.html>`_) this is supported both as member functions and using the NumPy functions. For example, using NumPy::
+
+   >>> r3 = odl.Rn(3)
+   >>> x = r3.element([1, 2, 3])
+   >>> np.negative(x)
+   Rn(3).element([-1.0, -2.0, -3.0])
+
+This method of using NumPy always uses the NumPy implementation, which can occur overhead in case the data is not stored in a CPU space. To always enable optimized code, users can call the member `NtuplesBaseVector.ufunc`::
+
+   >>> x.ufunc.negative()
+   Rn(3).element([-1.0, -2.0, -3.0])
+
+For other arbitrary functions, ODL vectors are generally accepted as input, but the output is often of `numpy.ndarray` type::
+
+   >>> np.convolve(x, x, mode='same')
+   array([  4.,  10.,  12.])
+
+
+NumPy functions as Operators
+----------------------------
+To solve the above issue, it is often useful to write an `Operator` wrapping NumPy functions, thus allowing full access to the ODL ecosystem. To wrap the convolution operation, a user could write a new class::
+
+   class MyConvolution(odl.Operator):
+       """Operator for convolving with given vector"""
+
+       def __init__(self, vector):
+           """Create a Convolution by vector"""
+           self.vector = vector
+
+           # Initialize operator superclass. This is a operator from the space
```

Comment:
Hm, I *thought* I knew where I caught this up, but I can't remember properly. At least the famous [super() considered super](https://rhettinger.wordpress.com/2011/05/26/super-considered-super) blog post (and related talks) never use the notion of a superclass. It's okay to use here because there really only is one "superclass", but otherwise, people usually talk about *the* superclass, which is not correct in Python. Parent class is always okay.